MGMT-8444: Update edit hostname dialog

rawagner opened the pull request:
![change_hostname](https://user-images.githubusercontent.com/2078045/148062742-8cbf3455-82f9-4bd1-be05-ed1571501bf1.gif)
![Screenshot from 2022-01-04 14-04-33](https://user-images.githubusercontent.com/2078045/148063665-c2a54e14-ff17-4908-83a3-133379d0515c.png)


![Screenshot from 2022-01-04 14-04-21](https://user-images.githubusercontent.com/2078045/148063414-9538571b-7746-44e7-8618-65fb76849147.png)
![Screenshot from 2022-01-04 14-06-23](https://user-images.githubusercontent.com/2078045/148063602-c6564e41-0ed0-4714-8c3f-f3e092fcd1dc.png)




@jkilzi FYI this also updates the dialog in OCM

ACM PR https://github.com/open-cluster-management/console/pull/1065

jkilzi commented:
@rawagner lgtm, but...
@nirfarkas are you aware of this change? Are you happy with it?

nirfarkas commented:
Not really ):
Why not change to using this?
https://github.com/patternfly/patternfly/issues/4017
It's interactive takes less space and is more elegant.

ShiranHi commented:
Using this rich helper text: https://marvelapp.com/prototype/haechg7/screen/84022338
Also in the Jira ticket: https://issues.redhat.com/browse/MGMT-8444

jkilzi commented on src/common/components/hosts/EditHostForm.tsx:
```diff
@@ -104,14 +108,26 @@ const EditHostForm: React.FC<EditHostFormProps> = ({
                 name="hostname"
                 ref={hostnameInputRef}
-                helperText="This name will replace the original discovered hostname after installation."
+                helperText={
+                  <List className="edit-hostname__help-text">
+                    <ListItem>Must be at least 5 characters</ListItem>
+                    <ListItem>Use lower-case letters, digits from 0 to 9 or hyphen (-)</ListItem>
+                    <ListItem>Cannot start or end with a hyphen (-)</ListItem>
+                    <ListItem>Cannot be the word "localhost"</ListItem>
+                  </List>
+                }
```
@ShiranHi Patternfly already provides dynamic helper texts.
Please see this example: https://www.patternfly.org/v4/components/helper-text#dynamic-list

jkilzi commented on src/common/components/hosts/EditHostForm.tsx:
```diff
@@ -104,14 +108,26 @@ const EditHostForm: React.FC<EditHostFormProps> = ({
                 name="hostname"
                 ref={hostnameInputRef}
-                helperText="This name will replace the original discovered hostname after installation."
+                helperText={
+                  <List className="edit-hostname__help-text">
+                    <ListItem>Must be at least 5 characters</ListItem>
+                    <ListItem>Use lower-case letters, digits from 0 to 9 or hyphen (-)</ListItem>
+                    <ListItem>Cannot start or end with a hyphen (-)</ListItem>
+                    <ListItem>Cannot be the word "localhost"</ListItem>
+                  </List>
+                }
```
@rawagner, @ShiranHi Patternfly already provides dynamic helper texts.
Please see this example: https://www.patternfly.org/v4/components/helper-text#dynamic-list

jkilzi replied:
We have already a task in order to align the PF spec in this regard.
See: https://issues.redhat.com/browse/MGMT-8275

rawagner commented:
allright, i misunderstood what this rich text component should represent. Sorry about that.
PR is now updated (pics included). 
@ShiranHi can you take a look again ? I've also added/changed some validation messages to properly reflect what we validate

nirfarkas commented:
@rawagner are the pics here updated?

ShiranHi commented:
Thanks @rawagner it looks good, I am happy to see it :)
Just wrapping up our call: "Must be unique" will be the first bullet point and we use the custom check and cross marks icons, as in [PF](https://www.patternfly.org/v4/components/helper-text/#dynamic-list).

jkilzi commented:
@rawagner we can really make use of a GIF here showing how it behaves

rawagner commented:
@nirfarkas I did not press `Update comment` ..so ehm no, screenshots were not updated :) 

Anyways I added a gif as @jkilzi suggested.

@ShiranHi 
- Moved `Must be unique` to top
- Added 'default' state when there is no value in the field. Note that the default state does not have bullets but minus icon since there is no option to have bullets in PF react
